src/commonMain/kotlin/com/vhl/blackmo/grass/pot/Root.kt

```diff
@@ -8,56 +8,34 @@ import kotlin.reflect.*
 /**
  * @author blackmo18
  */
-@Suppress("UNCHECKED_CAST")
 @ExperimentalStdlibApi
-abstract class Root<out T>(
-    private val type: KClass<*>,
-    private val trim: Boolean,
-    private val receivedKeyMap: Map<String, String>?
+open class Root<out T>(
+        val type: KClass<*>,
+        private val trim: Boolean,
+        val receivedKeyMap: Map<String, String>?
 ) {
 
-    private val paramNTypes = mutableMapOf<String?, ((String) -> Any)? >()
-    private val paramNIndex = mutableMapOf<String?, Int >()
-    private val paramNames = mutableListOf<String?>()
-    private val processedKey = mutableSetOf<String>()
-    private val customKeyMap = mutableMapOf<String,String>()
+    /**
+     * Key-value pair containing the expression fro converting from from csv column
+     * to actual type(class definition)
+     */
+    protected val paramNTypes = mutableMapOf<String?, ((String) -> Any)? >()
 
-    private fun initOnMethod() {
-        type.constructors.first().parameters.forEach {
-            paramNTypes[it.name] = getType(it.type)
-            paramNIndex[it.name] = it.index
-            paramNames.add(it.name)
-            receivedKeyMap?.let {
-                customKeyMap.putAll(it)
-            }
-        }
-    }
+    /**
+     * Key value pair index or oder of the csv column
```

Comment:
fix spelling `oder` to `order`
```suggestion
     * Key value pair index or oder of the csv column
```